tests/checkout/crlf.c

```diff
@@ -107,6 +108,9 @@ static void test_checkout(const char *autocrlf, const char *attrs)
 
 	git_buf_puts(&reponame, "crlf");
 
+	git_buf_puts(&systype_and_direction, systype);
+	git_buf_puts(&systype_and_direction, "_to_workdir");
```

Comment:
We definitely need to check return values for these here. Otherwise, if we run into an OOM, then we'll continue just fine and do the wrong thing later on. Same for all the other, pre-existing buffer operations

Reply:
Fixed.